### DISCUSSION TITLE
Add SurrealDB template

### DISCUSSION THREAD

**Discussion by sandros94** [coollabsio/coolify#4013] [Database Requests]
Add a template for [SurrealDB](https://surrealdb.com) ([their Github repo](https://github.com/surrealdb/surrealdb))
> A scalable, distributed, collaborative, document-graph database, for the realtime web.

I'm opening this after canceling of the related PR (https://github.com/coollabsio/coolify/pull/2617).
Mainly because of personal lack in php as well as a dependency for the https://github.com/coollabsio/coolify/discussions/3976 feature request that add support for SSL to Databases. This is because SurrealDB should be exposed to the public by default, considering that all its communications evolve around HTTP/RPC endpoints (including those coming from the end-users).

Some of its key features to keep in consideration while adding it (all are controllable both via [env variables](https://surrealdb.com/docs/surrealdb/cli/env) or [via args](https://surrealdb.com/docs/surrealdb/cli/start)):
- Supports both system (database) and access (end-user) authentication. Also via external OAuth.
- Ability to store files locally (via RocksDB or SurrealKV), or remotely (FoundationDB, TiKV).
- Ability to enable or disable various capabilities (network requests, functions, scripts, public access).
- Requires a `user` and a `pass` during first creation for root.

For those who want to use it before this gets implemented you can refer to the original [`docker-compose.yaml` from the PR](https://github.com/coollabsio/coolify/blob/42e171de2afd863baaa0aa4383c97b122400732e/templates/compose/surrealdb.yaml).

**Comment by ctwhome** [comment]
this just would be awesome to have it as an official template. I got it working thanks to your PR tho!

**Comment by horvbalint** [comment]
Hi, are there any plans to continue this effort of bringing SurrealDB to coolify (out of the box)?
I see that the SSL PR mentioned as a blocker got merged. + I think it would be great if the user could decide whether to expose the db publicly or not, like with MongoDB.
